test/wallet-test.js

```diff
@@ -2420,6 +2442,135 @@ describe('Wallet', function() {
       assert.strictEqual(bal.ulocked, value);
       assert.strictEqual(bal.clocked, value);
     });
+
+    it('should add external REVEAL to txdb', async () => {
+      // The goal is to have a "second-highest" bid
+      // so the wallet doesn't win the name for free.
+      // We can skip the whole BID/lockup thing for these tests.
+
+      const output = new Output();
+      output.value = secondHighest;
+      output.covenant.type = types.REVEAL;
+      output.covenant.pushHash(nameHash);
+      output.covenant.pushU32(start);
+      output.covenant.push(Buffer.from(name, 'ascii'));
+      output.covenant.pushHash(Buffer.alloc(32));
+
+      const mtx = new MTX();
+      mtx.outputs.push(output);
+
+      // Confirm external REVEAL
+      const block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(mtx.toTX(), block);
+    });
+
+    it('should confirm new REGISTER', async () => {
+      // Advance to close
+      wdb.height += network.names.revealPeriod;
+
+      const resource = Resource.fromJSON({records: []});
+      const register = await wallet.createUpdate(name, resource, {hardFee: fee});
+
+      // Check
+      let bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 4);
+      assert.strictEqual(bal.coin, 4);
+      assert.strictEqual(bal.confirmed, 10e6 - (3 * fee));
+      assert.strictEqual(bal.unconfirmed, 10e6 - (3 * fee));
+      assert.strictEqual(bal.ulocked, value);
+      assert.strictEqual(bal.clocked, value);
+
+      // Confirm REGISTER
+      const block = {
+        height: wdb.height + 1,
+        hash: Buffer.alloc(32),
+        time: Date.now()
+      };
+      await wallet.txdb.add(register.toTX(), block);
+
+      // Check
+      bal = await wallet.getBalance();
+      assert.strictEqual(bal.tx, 5);
+      // Wallet coin count doesn't change:
+      // REVEAL + fee money -> REGISTER + change
+      assert.strictEqual(bal.coin, 4);
+      assert.strictEqual(bal.confirmed, 10e6 - (4 * fee));
```

Comment:
Maybe it would be more scalable to have a global "transaction count" variable and a helper function `feeDiff` or `spentFees` or something like that, so that each section doesn't need to do `(n * fee)` and instead you could just call a function that increments the global variable while returning the required number. What do you think?

Reply:
Ok, that's what we did in #464 but not in #387. I think you're right though its a good precedent to set and looks cleaner, will update.